### ISSUE TITLE
General Model Design

### ISSUE THREAD

**Issue by metaloph1l** [metaloph1l/pacman#1]
Design has to contain the labyrinth, its components and the pacmans.

**Comment by lanoxx** [comment]
I dont think we should make sparate fields for walls and squares. There are only squares and each square has a flag that says which side has a wall, we then draw the walls onto the square according to these flags, or better we create a set of 10 different svg files (top, bottom, left, right, topBottom, leftRight, topLeftRight, bottomLeftRight, topLeftBottom, bottomRightTop) and then load these images accoring to the flags of the square.

**Comment by metaloph1l** [comment]
Yeah, thats a point that is up for discussion. There are of course these two possibilities:
1. Map the walls as separate objects
2. Map the walls as variations of a normal tile

The problem I have with the second approach is consistency. If you have a tile that has a wall on its right side the right neighbour needs a wall on its left side. Otherwise you would always have to check both the current and the neighbour tile. With the first approach you only have to check if the neighbour tile is a wall.

Another problem I see is that in the first case we would have many many types of different tiles {NO_WALL, WALL_UP, WALL_UP_DOWN, WALL_UP_LEFT, etc.} whereas in the first approach we would only have to different tile types. In the latter we would normally have large switches (with many cases).

The second option would therefore result in two large switches (both own tile and neighbour tile), whereas in the first approach we could just use a single if-else block. I think it would be easier that way.

**Comment by lanoxx** [comment]
That is a valid argument. Also the original Pacman design seems to have used walls as separate objects: http://www.google.com/pacman/

So we would add a boolean to the square field `isWall` an then check this when moving to a new field. If the next field would be a wall we set the direction in the pacman to `NONE` and stay on the current field.

**Comment by lanoxx** [comment]
I have created a domain branch with an initial commits. Feel free to add to it and if everyone agrees then we can merge it into master. Note that the name model is a very old term for domain so i renamed it to domain.

**Comment by metaloph1l** [comment]
Well, I guess next time I'll upload my code right away, because my code from yesterday just got obsolete.... :(

**Comment by lanoxx** [comment]
Well you can commit it to a separate branch so we can have a look at it. If its better than mine we can also take yours.

**Comment by spookyTU** [comment]
this is probably a little bit late on this issue but I also think that separate objects are easier. In this case a simple check if the next field is a wall is enough and we would be more flexible in adding "new kinds of walls"

**Comment by lanoxx** [comment]
Yes I agree with that. The arguments for separate objects are better than my idea with flags
